packages/medusa/src/strategies/batch-jobs/order/export.ts

```diff
@@ -0,0 +1,310 @@
+                message: `There will be ${count} orders exported by this action`,
+              },
+            ],
+          },
+        })
+    })
+  }
+
+  async processJob(batchJobId: string): Promise<void> {
+    let offset = 0
+    let limit = this.DEFAULT_LIMIT
+    let advancementCount = 0
+    let orderCount = 0
+    let approximateFileSize = 0
+
+    return await this.atomicPhase_(
+      async (transactionManager) => {
+        let batchJob = (await this.batchJobService_
+          .withTransaction(transactionManager)
+          .retrieve(batchJobId)) as OrderExportBatchJob
+
+        const { writeStream, fileKey, promise } = await this.fileService_
+          .withTransaction(transactionManager)
+          .getUploadStreamDescriptor({
+            name: `exports/orders/order-export-${Date.now()}`,
+            ext: "csv",
+          })
+
+        advancementCount =
+          batchJob.result?.advancement_count ?? advancementCount
+        offset = (batchJob.context?.list_config?.skip ?? 0) + advancementCount
+        limit = batchJob.context?.list_config?.take ?? limit
+
+        const { list_config = {}, filterable_fields = {} } = batchJob.context
+        const [, count] = await this.orderService_.listAndCount(
+          filterable_fields,
+          {
+            ...list_config,
+            order: { created_at: "DESC" },
+            skip: offset,
+            take: Math.min(batchJob.context.batch_size ?? Infinity, limit),
+          }
+        )
+
+        const lineDescriptor = this.getLineDescriptor(
+          list_config.select as string[],
+          list_config.relations as string[]
+        )
+
+        const header = this.buildHeader(lineDescriptor)
+        approximateFileSize += Buffer.from(header).byteLength
+        writeStream.write(header)
+
+        orderCount = batchJob.context?.batch_size ?? count
+
+        const result = batchJob.result
+        let orders = []
+
+        while (offset < orderCount) {
+          orders = await this.orderService_
+            .withTransaction(transactionManager)
+            .list(filterable_fields, {
+              ...list_config,
+              skip: offset,
+              take: Math.min(orderCount - offset, limit),
+            })
+
+          orders.forEach(async (order) => {
+            const line = await this.buildCSVLine(order, lineDescriptor)
+            approximateFileSize += Buffer.from(line).byteLength
+            writeStream.write(line)
+          })
+
+          batchJob = (await this.batchJobService_
+            .withTransaction(transactionManager)
+            .update(batchJobId, {
+              result: {
+                file_key: fileKey,
+                file_size: approximateFileSize,
+                count: orderCount,
+                advancement_count: advancementCount,
+                progress: advancementCount / orderCount,
+              },
+            })) as OrderExportBatchJob
+
+          advancementCount += orders.length
```

Comment:
Suggestion: if you move that above the update, you don’t need the update after the loop at the end since it will tale into account the last data wdyt? I did that for the product